### PR TITLE
[JENKINS-64393] update jupiter to fix not running junit4 tests

### PR DESCRIPTION
**Pull request by jtnord**
the version of jupiter was incompatible with the version of junit4
causing all junit4 tests to not run (and not fail).

(no tests as the tester isn't tested), but should be able to validate by comparing the CI tests# from master vs this PR.

See [JENKINS-64393](https://issues.jenkins-ci.org/browse/JENKINS-64393).

### Proposed changelog entries

N/A 

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: cli/pom.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
   <properties>
     <!-- TODO: remove once Medium level is set in the core (JENKINS-36716) -->
     <findbugs.threshold>Medium</findbugs.threshold>
-    <junit.jupiter.version>5.5.2</junit.jupiter.version>
+    <junit.jupiter.version>5.7.0</junit.jupiter.version>
```
Comment (jglick) [inline cli/pom.xml:19]: Can we move the version into the parent POM so it is managed in just one place?

**Review by jglick — APPROVED** [review]
FWIW locally tests do run despite the stack trace.

**Review by jvz — APPROVED** [review]

**Comment by jtnord** [comment]
> FWIW locally tests do run despite the stack trace.

not for me 🤷   at least on the test I picked at not random that was junit4 based. 
The test graph in Jenkins also seems to show a large drop off in tests. some tests run (I was assuming those where the junit5 based ones - but have not run the whole test suite to verify)

File: cli/pom.xml (hunk 1 above)
Comment (jtnord) [reply cli/pom.xml]: addressed in eff990fe05

**Comment by jglick** [comment]
Maybe add to `Jenkinsfile` e.g.:

```groovy
sh 'test -f core/target/surefire-reports/TEST-hudson.FilePathTest.xml'
```

**Review by timja — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

**Comment by timja** [comment]
Lots more tests passing compared to this one: https://github.com/jenkinsci/jenkins/pull/5103 :),
just reviewed the count from GitHub checks report

**Pull request merged**